packages/backend/src/links/dto/create-link.dto.ts

```diff
@@ -1 +1,31 @@
-export class CreateLinkDto {}
+import { ArrayMinSize, IsArray, IsNotEmpty, IsString, IsUUID, IsUrl } from 'class-validator';
+
+export class CreateLinkDto {
+  @IsUUID()
+  id: string;
+
+  @IsString()
+  @IsNotEmpty()
+  url: string;
+
+  @IsString()
+  @IsNotEmpty()
+  title: string;
+
+  @IsString()
+  @IsNotEmpty()
+  @IsUrl()
+  slug: string;
+
+  @IsString()
+  @IsNotEmpty()
+  description: string;
+
+  @IsString()
+  iconUrl: string;
```

Comment:
you can add the IsOptional() decorator here, I think otherwise it throws an error if the field is omitted in the request